Tests/ClientRuntimeTests/InterceptorTests/InterceptorTests.swift

```diff
@@ -0,0 +1,86 @@
+import ClientRuntime
+import XCTest
+
+class InterceptorTests: XCTestCase {
+    struct TestInput {
+        public var property: String?
+    }
+
+    struct TestOutput {
+        public var property: String?
+    }
+
+    struct AddAttributeInterceptor<T, RequestType, ResponseType, AttributesType: HasAttributes> : Interceptor {
+        private let key: AttributeKey<T>
+        private let value: T
+
+        init(key: AttributeKey<T>, value: T) {
+            self.key = key
+            self.value = value
+        }
+
+        public func modifyBeforeSerialization(context: inout some MutableInput<Self.AttributesType>) async throws {
+            let attributes = context.getAttributes()
+            attributes.set(key: self.key, value: self.value)
+        }
+    }
+
+    struct ModifyInputInterceptor<InputType, RequestType, ResponseType, AttributesType: HasAttributes> : Interceptor {
+        private let keyPath: WritableKeyPath<InputType, String?>
+        private let value: String
+
+        init(keyPath: WritableKeyPath<InputType, String?>, value: String) {
+            self.keyPath = keyPath
+            self.value = value
+        }
+
+        public func modifyBeforeSerialization(context: inout some MutableInput<Self.AttributesType>) async throws {
+            var input: InputType = context.getInput()!
+            input[keyPath: keyPath] = value
+            context.updateInput(updated: input)
+        }
+    }
+
+    struct AddHeaderInterceptor : HttpInterceptor {
```

Comment:
nit: Add a test interceptor that implements multiple interceptor hooks, e.g. both `modifyBeforeSerialization` and `modifyBeforeTransmit` in one definition